week-02/assessment/gomoku/src/main/java/learn/gomoku/App.java

```diff
@@ -1,7 +1,27 @@
 package learn.gomoku;
+import learn.gomoku.game.Gomoku;
+import learn.gomoku.players.Player;
+
 
 public class App {
 
     public static void main(String[] args) {
+        UiGomoku.welcomeMessage();
+        do {
+            Gomoku game = UiGomoku.startGame();
+
+            Player currentPlayer;
+
+            while (!game.isOver()){
+
+                currentPlayer = game.getCurrent();
+                UiGomoku.displayBoard(game);
+                UiGomoku.playerTurn(game, currentPlayer);
+            }
+            UiGomoku.displayBoard(game);
+
+        }while (UiGomoku.playAgain());
+
+        UiGomoku.endGame();
```

Comment:
This is far too much inside of your main app, and everything is statically being called, which makes this very much procedural in the way it's being handled.  Everything here should be moved to a run method.

```suggestion
        UiGomoku gomokuGame = new UiGomoku();
        gomokuGame.run();
```